openmoc/materialize.py

```diff
@@ -556,7 +566,8 @@ def compute_sph_factors(mgxs_lib, max_sph_iters=30, sph_tol=1E-5,
         fsrs_to_domains[fsr] = domain.getId()
         domains_to_fsrs[domain.getId()].append(fsr)
 
-        if domain.isFissionable():
+        if (sph_domains == None and domain.isFissionable()) or \
+             domain.getId() in sph_domains:
```

Comment:
This will create a problem when `sph_domains` is None. The loop could be rewritten something like below, `sph_domains` could be initialized elsewhere.

```python
        if not sph_domains:
            if domain.isFissionable():
                sph_to_fsr_indices.append(fsr)
        else:
            if domain.getId() in sph_domains:
                sph_to_fsr_indices.append(fsr)
```

Reply:
I like the second solution the best, implemented that way